cli/tests/test_next_prev_commands.rs

```diff
@@ -0,0 +1,223 @@
+    let stderr = test_env.jj_cmd_cli_error(&repo_path, &["next"]);
+    insta::assert_snapshot!(stderr,@r#""#);
+}
+
+#[test]
+fn test_prev_fails_on_multiple_parents() {
+    // TODO(#NNN): Fix this behavior
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "second"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "third"]);
+
+    // Try to access a parent commit.
+    let stderr = test_env.jj_cmd_cli_error(&repo_path, &["prev"]);
+    insta::assert_snapshot!(stderr,@r#""#);
+}
+
+#[test]
+fn test_prev_onto_root_fails() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "second"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "third"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "fourth"]);
+    // The root commit is before "first".
+    let stderr = test_env.jj_cmd_cli_error(&repo_path, &["prev", "4"]);
+    insta::assert_snapshot!(stderr,@r#""#);
+}
+
+#[test]
+fn test_prev_editing() {
+    // Edit the third commit.
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "second"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "third"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "fourth"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["prev", "--edit"]);
+    insta::assert_snapshot!(
+        stdout,
+        @r###"
+    Working copy now at: 009f88bf7141 fourth
+    Parent commit      : 3fa8931e7b89 third
+    "###
+    );
+}
+
+#[test]
+fn test_next_editing() {
+    // Edit the second commit.
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "second"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "third"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "fourth"]);
+    test_env.jj_cmd_success(&repo_path, &["edit", "@--"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["next", "--edit"]);
+    insta::assert_snapshot!(
+        stdout,
+        @r###"
+    Nothing changed.
+    "###
+    );
```

Comment:
Could you also add a test without the `--edit` here (but still the `edit @--` above)?

Reply:
That is `test_next_multiple` or am I missin gsomething.

Reply:
You're right.